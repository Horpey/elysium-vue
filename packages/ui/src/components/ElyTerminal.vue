<template>
  <div class="border border-l-8 border-koromiko-800 bg-koromiko-800/10 px-4 py-2">
    <slot />
  </div>
</template>
