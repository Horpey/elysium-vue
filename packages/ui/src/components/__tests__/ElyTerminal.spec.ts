import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import ElyTerminal from '../ElyTerminal.vue'

describe('elyTerminal', () => {
  it('renders correctly', () => {
    const wrapper = mount(ElyTerminal)

    // Assert that the component is rendered
    expect(wrapper.exists()).toBe(true)

    // Assert that slot content is not provided
    expect(wrapper.text()).toBe('')

    // Assert snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders slot content', () => {
    const wrapper = mount(ElyTerminal, {
      slots: {
        default: '<p>Slot Content</p>',
      },
    })

    // Assert that the component is rendered
    expect(wrapper.exists()).toBe(true)

    // Assert slot content is rendered
    expect(wrapper.find('p').text()).toBe('Slot Content')

    // Assert snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })
})
