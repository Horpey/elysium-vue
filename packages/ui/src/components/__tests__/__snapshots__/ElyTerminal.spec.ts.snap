// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`elyTerminal > renders correctly 1`] = `"<div class="border border-l-8 border-koromiko-800 bg-koromiko-800/10 px-4 py-2"></div>"`;

exports[`elyTerminal > renders slot content 1`] = `
"<div class="border border-l-8 border-koromiko-800 bg-koromiko-800/10 px-4 py-2">
  <p>Slot Content</p>
</div>"
`;
